### PR TITLE
polish up ML elaborator; new concrete syntax

### LINKED ISSUE #645: Change the syntax of Thm?
Use something like
```
Thm X in A by [ ... ]
```
or
```
Thm X : A by [ ... ]
```
instead of
```
Thm X : [ A ] by [ ... ]
```

### PR DESCRIPTION
**Pull request by jonsterling**
Resolves #645 

1. polish metalanguage elaborator
2. new concrete notation for RedPRL signatures
3. start parsing RedML code in RedPRL signatures directly; see `example/metalanguage.prl`

TODO:

- [x] update vim mode
- [x] update docs
- [x] update pygments lexer

### CODE REVIEW

File: example/J.prl [hunk 1]
```diff
@@ -12,7 +12,7 @@ define J/coe(#j:dim, #ty, #a, #fam, #d, #p) =
    #d)
 .
 
-theorem J(#l:lvl) :
+theorem J(#l:lvl) :
```
Comment (cangiuli) [inline example/J.prl]: jon pls

Comment (jonsterling) [reply example/J.prl]: shit!!

**Comment by favonia** [comment]
Should I wait for you before changing the parser?

**Comment by jonsterling** [comment]
@favonia I guess it depends on which PR we expect to be merged first. I think I can get this one ready to merge today.

File: src/redprl/metalanguage/elaborate.fun [hunk 2]
```diff
@@ -196,6 +196,16 @@ struct
       (ISyn.RET v, Ty.UP vty)
     end
 
+  (* Expects a multitactic in 'script' *)
+  fun refineSequents (name, sequents : Sequent.jdg list, script : RedPrlAbt.abt) : icmd * cty =
```
[the remark this answers is not in the document]
Comment (favonia) [reply src/redprl/metalanguage/elaborate.fun]: Oh yeah!

**Comment by jonsterling** [comment]
@favonia How do you feel about merging this PR right away? That way you can then edit the parser without fear of running into conflicts...

@cangiuli I realize that my changes to the vim mode are very rudimentary and don't yet reflect the complete new grammar, but I think it at least works for existing code.

**Comment by favonia** [comment]
> How do you feel about merging this PR right away?

You have my support.

**Comment by jonsterling** [comment]
@favonia could you approve the pr then?

**Pull request merged**